### PR TITLE
Removing library global variables and also fixing some minor problems

### PR DESCRIPTION
**Pull request by sclaomir**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -31,17 +31,19 @@ set(SRC_FTI
 append_property(SOURCE ${SRC_FTI}
 	PROPERTY COMPILE_FLAGS "${MPI_C_COMPILE_FLAGS}")
 
-add_library(fti.static STATIC ${SRC_FTI})
-target_link_libraries(fti.static ${MPI_C_LIBRARIES} iniparser jerasure)
-
-add_library(fti.shared SHARED ${SRC_FTI})
-target_link_libraries(fti.shared ${MPI_C_LIBRARIES} iniparser jerasure)
```
Comment (jbigot) [inline CMakeLists.txt]: it looks like the MPI libraries have been removed. Is that on purpose?

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,10 +1,10 @@
-cmake_minimum_required(VERSION 2.8 FATAL_ERROR)
+cmake_minimum_required(VERSION 2.8.8 FATAL_ERROR)
 
 if("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_CURRENT_SOURCE_DIR}")
 	project(FTI C Fortran)
 endif()
 
-option(ENABLE_FORTRAN "Enables the generation of the Fortran wrapper for FTI" ON)
+option(ENABLE_FORTRAN "Enables the generation of the Fortran wrapper for FTI" OFF)
```
Comment (sclaomir) [inline CMakeLists.txt]: by accident

**Comment by leobago** [comment]
Slawek did you test this code using a dedicated thread for checkpointing? I have done 5 tests in this configuration and none of them worked properly. The application does not crash but the checkpoint directory renaming is not managed properly and the checkpoints accumulate in the tmp directory.

**Comment by sclaomir** [comment]
I was checking with and without dedicated process. And I was analyzing only library output and not directories tree. So indeed in the output there is no warning or error but renaming isn't working properly.

But this problem is much older.. I checked master branch and this problem is also present there. Looking into code I found only one place in which it is decided which rank should rename directories (checkpoint.c: FTI_PostCkpt)
`
nodeFlag = 
(((!FTI_Topo->amIaHead) && (FTI_Topo->nodeRank == 0)) || (FTI_Topo->amIaHead)) ? 1 : 0;
`
It looks like that this condition is always false when using dedicated process. 

I ran some more tests to check this problem (both on local machine and on cluster). And all ranks which are calling `FTI_PostCkpt` function always have `amIaHead` equal `0`.

**Comment by leobago** [comment]
Right, nodeFlag is the variable that is supposed to tell which process takes care of the cleaning in the node. When there is no dedicated process is nodeRank 0 who takes the lead. When there is dedicated processes (heads), the y should be in charge of the cleaning. FTI_Topo->amIaHead should be 1 for dedicated processes and then get nodeFlag = 1. Can you check the value of amIaHead?

**Comment by sclaomir** [comment]
Something strange is going on :) I ran example on 4 nodes each 28 cores. After `FTI_Init` ranks 0, 28, 56 and 84 have `amIaHead` set to 1. But later `FTI_PostCkpt` function is called in many ranks, but not in rank 0, 28, 56 and 84.

**Comment by leobago** [comment]
Yes that is weird. Are you sure none of the head processes is calling FTI_PostCkpt? It is supposed to be called from the FTI_Listen function.

**Comment by sclaomir** [comment]
Yes, I'm sure. Moreover, all ranks except "heads" are calling `FTI_PostCkpt`.

**Comment by leobago** [comment]
I got it. It is normal `FTI_PostCkpt` is not called by the heads. This is because we set `isInline` to 1 which means the application process should do the PostCkpt work by itself. Which is also a correct behaviour. The only problem is that once inside the PostCkpt function the current nodeFlag test does not takes this scenario into account. We have to fix that test.

**Comment by leobago** [comment]
@sclaomir I just pushed a fix into my devel branch, could you please double check that this fix the bug? Thanks.

**Comment by sclaomir** [comment]
It seems that now it works correctly.. 

One remark. When inlining is set to 0 for levels L2, L3 and L4, no-head ranks taking these checkpoints print warnings
`[FTI Warning 000001] : FTI failed to take checkpoint.`
`[FTI Warning 000001] : Error => No such file or directory `
or
`[FTI Warning 000005] : FTI failed to take checkpoint. `
`[FTI Warning 000005] : Error => File exists `
but all checkpoints are created properly.

**Comment by leobago** [comment]
Yes, Mohamed noticed that yesterday. Any idea what is going on?

**Comment by sclaomir** [comment]
So, in `FTI_Checkpoint` function `res` is set to `res = FTI_REJW;` or `res = FTI_BASE + FTI_Exec.ckptLvel;` and then this value is sent to the head rank. This is done for not-inline branch. After that `res` is modified one more time before exit and set to `res = FTI_DONE;` if it was earlier equal to `FTI_SCES`. But this case is valid only for inline branch. 

I will provide fix in a moment.

**Comment by leobago** [comment]
You are right @sclaomir ! We just need to change the test to `if (res != FTI_NSCS)`
I will let you do it and send the pull request.

**Pull request merged**